lib/postprocessing/watermarkProcessor.js

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+'use strict';
+
+const { getDimensions } = require('./assetProperties');
+const { ImageOperations } = require('./imageOperations');
+
+class WatermarkProcessor {
+    constructor(instructions) {
+
+        this.renditionWidth = instructions.width || this.getRenditionDimensions(instructions.path).width || 0;
+        this.renditionHeight = instructions.height || this.getRenditionDimensions(instructions.path).height || 0;
+        this.watermarkWidthPercent = instructions.watermark.widthPercent || 100;
+        this.watermarkPath = instructions.watermark.path;
+        this.getWatermarkDimensions(this.watermarkPath);
+    }
+
+    getRenditionDimensions (path) {
+        return getDimensions(path);
+    }
+
+    getWatermarkDimensions (path) {
+        const dimensions = getDimensions(path);
+        this.watermarkWidth = dimensions.width;
+        this.watermarkHeight = dimensions.height;
+    }
+
+    compareRenditionWidth () {
+        console.log("renditionWidth" + this.renditionWidth + "<=" +  "watermarkWidth" + this.watermarkWidth );
```

Comment:
Nitpick 1: 1 space too much at the end before the parenthesis - or one missing at the beginning.
Nitpick 2: Why a concat and not the string interpolation (`${my-var}... `)?

Reply:
This is a debug line